File: ProductPurchaseApp/ProductPurchaseApp.Web/Modules/ProductPurchaseRequestDB/ProductPurchaseRequest/ProductPurchaseRequestForm.cs
using Serenity.ComponentModel;
using System;

namespace ProductPurchaseApp.ProductPurchaseRequestDB.Forms;

[FormScript("ProductPurchaseRequestDB.ProductPurchaseRequest")]
[BasedOnRow(typeof(ProductPurchaseRequestRow), CheckNames = true)]
public class ProductPurchaseRequestForm
{
    //[ReadOnly(true)]
    //public int UserId { get; set; }
    public string ProductName { get; set; }
    public string Status { get; set; }
    [MultipleFileUploadEditor(FilenameFormat ="products/attachment/~")]
    public string Attachment { get; set; }
    [DateEditor]
    public DateTime CreatedDate { get; set; }
}

File: ProductPurchaseApp/ProductPurchaseApp.Web/Modules/ServerTypes/ProductPurchaseRequestDB/ProductPurchaseRequestForm.ts
import { StringEditor, MultipleImageUploadEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ProductPurchaseRequestForm {
    ProductName: StringEditor;
    Status: StringEditor;
    Attachment: MultipleImageUploadEditor;
    CreatedDate: DateEditor;
}

export class ProductPurchaseRequestForm extends PrefixedContext {
    static readonly formKey = 'ProductPurchaseRequestDB.ProductPurchaseRequest';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProductPurchaseRequestForm.init)  {
            ProductPurchaseRequestForm.init = true;

            var w0 = StringEditor;
            var w1 = MultipleImageUploadEditor;
            var w2 = DateEditor;

            initFormType(ProductPurchaseRequestForm, [
                'ProductName', w0,
                'Status', w0,
                'Attachment', w1,
                'CreatedDate', w2
            ]);
        }
    }
}